backend/app/services/diagnosis_agent.py

```
@@ -32,462 +37,552 @@ class DiagnosisState(TypedDict):
+                
+            if row:
+                parts = []
+                if row.ecology_info: parts.append(f"### {pest} 생태정보\n\n{row.ecology_info.strip()}")
+                if row.prevent_method: parts.append(f"### {pest} 방제방법\n\n{row.prevent_method.strip()}")
+                return {"ncpms_data": "\n\n".join(parts) if parts else "데이터 없음"}
+            return {"ncpms_data": "NCPMS 데이터를 찾을 수 없습니다."}
+    except Exception:
+        logger.exception("NCPMS DB search error for pest %s", pest)
+        return {"ncpms_data": "NCPMS 조회 중 오류가 발생했습니다."}
 
+async def fetch_pesticide(state: DiagnosisState) -> dict:
```

Ruff (0.15.10)

[warning] Too many branches (15 > 12)

(PLR0912)